smack-extensions/src/main/java/org/jivesoftware/smackx/message_correct/MessageCorrectExtension.java

```diff
@@ -100,21 +100,8 @@ public String getNamespace() {
         return NAMESPACE;
     }
 
-    /**
-     * A ExtensionElementProvider for the MessageCorrectExtension. 
-     * As MessageCorrection elements have only the JID of the message to replace.
-     * 
-     * @author Fernando Ramirez, f.e.ramirez94@gmail.com
-     */
-    public static class Provider extends ExtensionElementProvider<MessageCorrectExtension> {
-
-        @Override
-        public MessageCorrectExtension parse(XmlPullParser parser, int initialDepth)
-                throws XmlPullParserException, IOException, SmackException {
-            String jidMessageToReplace;
-            jidMessageToReplace = parser.getAttributeValue("", JID_TAG);
-            return new MessageCorrectExtension(jidMessageToReplace);
-        }
+    public static MessageCorrectExtension from(Message message) {
+        return new MessageCorrectExtension(message.getStanzaId());
```

Comment:
It appears I was unclear with what `from(Message)` should do: Instead of creating a new MessageCorrect, it should extract the extension from the message if there is one. The body of the function should be
`return message.getExtension(ELEMENT, NAMESPACE);`